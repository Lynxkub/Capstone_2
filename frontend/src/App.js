import React , {useState , useEffect} from 'react';
import './App.css';
import Routes from './Routes';
import NavBar from './NavBar';
import FoodlyApi from './api';


function App() {

  const [userToken , setUserToken] = useState(localStorage.getItem('token'));
  const [username , setUsername] = useState(localStorage.getItem('username'));


  useEffect(() => {
    localStorage.setItem('token' , userToken)
  } , [userToken])


  FoodlyApi.token = localStorage.getItem('token');



  const signUp = async (newUser) => {
    const generatedToken = await FoodlyApi.register(newUser);
    setUserToken(generatedToken);
    localStorage.setItem('token' , generatedToken);
    setUsername(localStorage.getItem('username'));
  }

  const login = async (returningUser) => {
    const generatedToken = await FoodlyApi.authenticate(returningUser);
    setUserToken(generatedToken)
    localStorage.setItem('token' , FoodlyApi.token);
    setUsername(localStorage.getItem('username'));
  }


  const logout = () => {
    setUserToken(undefined);
    setUsername(localStorage.getItem('username'));
  }

  const addComment = async (comment) => {
    const createComment = await FoodlyApi.makeComment(comment);
    return createComment;
  }

  

  
  return (
    <div className="fill-window">
      <NavBar />
    <Routes signUp = {signUp} token = {userToken} login = {login}  logout = {logout} addComment = {addComment}/>
    </div>
  );
}

export default App;