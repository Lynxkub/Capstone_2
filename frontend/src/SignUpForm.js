import React , {useState} from 'react';
import { useHistory } from 'react-router-dom';


const SignUpForm = ( { signUp , token }) => {
    const history = useHistory();
    const INITIAL_STATE = {
        username : '',
        password : '',
        first_name : '',
        last_name : '',
        email : ''
    }

    const [formData , setFormData] = useState(INITIAL_STATE)

    const username = localStorage.getItem('username');

    if(username && username !== "") {
        history.push('/');
    }

    const handleChange = (e) => {
        const {name , value } = e.target;
        setFormData(formData => ({
            ...formData ,
            [name] : value
        }))
    }

    const handleSubmit = async (e) => {
        try{
            e.preventDefault();
            localStorage.setItem('username' , formData.username)
             await signUp(formData);
            setFormData(INITIAL_STATE)
            history.push('/');
        }catch(err) {
            alert(err);
        }
    }

    return (
        <div>
            <form onSubmit = {handleSubmit}>
                <label htmlFor = 'username'>Username</label>
                <input
                type = 'text'
                placeholder = 'Username'
                name = 'username'
                data-testid = 'username'
                value = {formData.username}
                onChange = {handleChange}
                />
                  <label htmlFor = 'password'>Password</label>
                <input
                type = 'password'
                placeholder = 'Password'
                name = 'password'
                data-testid = 'password'
                value = {formData.password}
                onChange = {handleChange}
                />
                  <label htmlFor = 'first_name'>First Name</label>
                <input
                type = 'text'
                placeholder = 'First Name'
                name = 'first_name'
                data-testid = 'first_name'
                value = {formData.first_name}
                onChange = {handleChange}
                />
                  <label htmlFor = 'last_name'>Last Name</label>
                <input
                type = 'text'
                placeholder = 'Last Name'
                name = 'last_name'
                data-testid = 'last_name'
                value = {formData.last_name}
                onChange = {handleChange}
                />
                  <label htmlFor = 'email'>Email</label>
                <input
                type = 'email'
                placeholder = 'Email'
                name = 'email'
                data-testid = 'email'
                value = {formData.email}
                onChange = {handleChange}
                />
                <button data-testid = 'signupButton'>Sign Up</button>
            </form>
        </div>
    )
}


export default SignUpForm;