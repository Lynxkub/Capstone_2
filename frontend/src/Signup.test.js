import React from 'react';
import {render , fireEvent} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import SignUpForm from './SignUpForm';
import App from './App';
import FoodlyApi from './api';

test('it renders without crashing' , () => {
    render(
        <MemoryRouter>
            <SignUpForm />
        </MemoryRouter>
    )
})


test('it matches snapshot' , () => {
    const { asFragment } = render (
        <MemoryRouter>
            <SignUpForm />
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot();
})


test('sign up for successfully signs up a new user' , () => {
    const {getByText , getByTestId} = render(
        <MemoryRouter initialEntries={['/']}>
            <App />
        </MemoryRouter>
    )

    const signupLink = getByText('Sign Up');
    fireEvent.click(signupLink);

    // make sure the link takes user to a sign up form

    const username = getByTestId('username');
    expect(username).toBeInTheDocument();

    // fill out information and submit

    const password = getByTestId('password');
    const email = getByTestId('email');
    const firstName = getByTestId('first_name');
    const lastName = getByTestId('last_name');
    const signUpButton = getByTestId('signupButton');

    fireEvent.change(username , {target : {value : 'newUser'}})
    fireEvent.change(password , {target : {value : 'password'}})
    fireEvent.change(firstName , {target : {value : 'newUser'}})
    fireEvent.change(lastName , {target : {value : 'newUser'}})
    fireEvent.change(email , {target : {value : 'newUser@newUser.com'}})
    fireEvent.click(signUpButton);

    setTimeout(async () => {
        const userData = await FoodlyApi.getUser('newUser');
        expect(userData).toEqual({
            username : 'newUser',
            first_name : 'newUser',
            last_name : 'newUser',
            email : 'newUser@newUser.com'
        } , 1000)
    })

})