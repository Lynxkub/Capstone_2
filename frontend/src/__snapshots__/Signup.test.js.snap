// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="username"
      >
        Username
      </label>
      <input
        data-testid="username"
        name="username"
        placeholder="Username"
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        data-testid="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        for="first_name"
      >
        First Name
      </label>
      <input
        data-testid="first_name"
        name="first_name"
        placeholder="First Name"
        type="text"
        value=""
      />
      <label
        for="last_name"
      >
        Last Name
      </label>
      <input
        data-testid="last_name"
        name="last_name"
        placeholder="Last Name"
        type="text"
        value=""
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        data-testid="email"
        name="email"
        placeholder="Email"
        type="email"
        value=""
      />
      <button
        data-testid="signupButton"
      >
        Sign Up
      </button>
    </form>
  </div>
</DocumentFragment>
`;
